vscode-extension/src/extension.ts

```diff
@@ -1,27 +1,290 @@
+		});
+	});
+
+	const devboxInit = commands.registerCommand('devbox.init', async () => {
+		ensureTerminalExists();
+		await commands.executeCommand('workbench.action.terminal.sendSequence', {
+			'text': 'devbox init\r\n'
+		});
+		commands.executeCommand('setContext', 'devbox.configFileExists', true);
 	});
+
+	context.subscriptions.push(devboxAdd);
+	context.subscriptions.push(devboxInit);
+	context.subscriptions.push(devboxShell);
+	context.subscriptions.push(devboxInstall);
+	context.subscriptions.push(setupDevcontainer);
 }
 
-async function runDevboxShell() {
+async function initialCheckDevboxJSON() {
+	// check if there is a workspace folder open
+	if (workspace.workspaceFolders) {
+		const workspaceUri = workspace.workspaceFolders[0].uri;
+		try {
+			// check if the folder has devbox.json in it
+			await workspace.fs.stat(Uri.joinPath(workspaceUri, "devbox.json"));
+			// devbox.json exists setcontext for devbox commands to be available
+			commands.executeCommand('setContext', 'devbox.configFileExists', true);
 
-	const result = await vscode.workspace.findFiles('devbox.json');
+			if (workspace.getConfiguration("devbox").get("promptUpdateSettings")) {
+				const response = await window.showInformationMessage(
+					"A Devbox project is opened. Do you want to project settings with Devbox environment?",
+					"Update Settings",
+					"Don't show again"
+				);
+				if (response === "Update Settings") {
+					const devboxJson = await readDevboxJson(workspaceUri);
+
+					updateSettings(workspaceUri.path, devboxJson);
+				} else if (response === "Don't show again") {
+					workspace.getConfiguration("devbox").update("promptUpdateSettings", false);
+				}
+			}
+		} catch (err) {
+			console.log(err);
+			// devbox.json does not exist
+			commands.executeCommand('setContext', 'devbox.configFileExists', false);
+			console.log("devbox.json does not exist");
+		}
+	}
+}
+
+function updateSettings(workspacePath: String, devboxJson: any) {
+	// Updating process.env.PATH
+	process.env["PATH"] = process.env["PATH"] + ":" + workspacePath + "/.devbox/nix/profile/default/bin";
+	// Updating language extension settings
+	// For now we only update Go, Python3, and Nodejs language extensions
+	devboxJson["packages"].forEach((pkg: String) => {
+		if (pkg.startsWith("python3")) {
+			if (extensions.getExtension("ms-python.python")?.isActive) {
+				workspace.getConfiguration("python").update("defaultInterpreterPath", workspacePath + "/.devbox/nix/profile/default/bin/python3");
+			}
+		}
+		if (pkg.startsWith("go_1_") || pkg === "go") {
+			if (extensions.getExtension("golang.go")?.isActive) {
+				workspace.getConfiguration("go").update("gopath", workspacePath + "/.devbox/nix/profile/default/bin/go");
+			}
+		}
+		if (pkg.startsWith("nodejs-") || pkg === "nodejs") {
+			if (extensions.getExtension("eslint")?.isActive) {
+				workspace.getConfiguration("eslint").update("nodepath", workspacePath + "/.devbox/nix/profile/default/bin/node");
```

Comment:
Thoughts: This part of the logic makes me nervous. I'd rather avoid it and ask the user to manually configure for now, or expose an API so that other extensions can invoke

Reply:
this is removed along with others parts for updating other extension settings